sdks/python/apache_beam/transforms/trigger.py

```diff
@@ -1064,6 +1064,18 @@ def __init__(self, defensive_copy=True):
     self.global_state = {}
     self.defensive_copy = defensive_copy
 
+  def copy(self):
+    cloned_object = copy.copy(self)
+    cloned_object.timers = copy.deepcopy(self.timers)
+    cloned_object.global_state = copy.deepcopy(self.global_state)
+    cloned_object.state = (
+        collections.defaultdict(lambda: collections.defaultdict(list)))
+    for window in self.state:
+      cloned_object.state[window] = collections.defaultdict(list)
```

Comment:
<!--new_thread; commit:3f1a714b613353fdfe62e46e63c7bbfa47d98295; resolved:0-->
You can omit this line--the dict is created for you by the defaultdict mechanism on `cloned_object.state`.